kidneys_misc.module

```diff
@@ -61,49 +66,214 @@ function kidneys_misc_form_alter(&$form, &$form_state, $form_id) {
+
+  // Setup fields for use below.
+  $fields_to_keep = $fields_true = $fields_false = array();
+  $address_field = $trial_redirect_path = $trial_location_field_name = '';
+
+  switch ($form['#bundle']) {
+    // Machine name of the entityform.
+    case 'iga':
+      // Adds machine name of fields whose data should be stored.
+      // All other fields data will be cleared after form processing.
+      $fields_to_keep = array(
+        'field_base_first_name',
+        'field_base_last_name',
+        'field_base_email',
+        'field_iga_closest_center',
+      );
+
+      // Add machine name of fields that should return TRUE.
+      $fields_true = array(
+        'field_iga_18',
+        'field_iga_diagnosis',
+        'field_iga_biopsy',
+      );
+
+      // Add machine name of fields that should return FALSE.
+      $fields_false = array(
+        'field_iga_cancer',
+        'field_iga_transplant',
+        'field_iga_end_stage',
+      );
+
+      // Machine name of field that stores user submitted address.
+      $address_field = 'field_base_address';
+
+      // Destination to be redirected to after form processing.
+      // The webform submission id will be appended to the URL when redirecting.
+      // Ex: iga-trial/results will redirect to iga-trial/results/430 where 430 is the submission id.
+      $trial_redirect_path = 'iga-trial/results';
+
+      // Machine name of field that should store the closest center found after form processing.
+      $trial_location_field_name = 'field_iga_closest_center';
+      break;
+// Duplicate, uncomment, and update values below to add a new trial.
+//    // Machine name of the entityform.
+//    case 'new_trial':
+//      // Adds machine name of fields whose data should be stored.
+//      // All other fields data will be cleared after form processing.
+//      $fields_to_keep = array(
+//        'field_base_first_name',
+//        'field_base_last_name',
+//        'field_base_email',
+//      );
+//
+//      // Add machine name of fields that should return TRUE.
+//      $fields_true = array(
+//        'field_new_trial_18',
+//      );
+//
+//      // Add machine name of fields that should return FALSE.
+//      $fields_false = array(
+//        'field_new_trial_cancer',
+//      );
+//
+//      // Machine name of field that stores user submitted address.
+//      $address_field = 'field_base_address';
+//
+//      // Destination to be redirected to after form processing.
+//      // The webform submission id will be appended to the URL when redirecting.
+//      // Ex: iga-trial/results will redirect to iga-trial/results/430 where 430 is the submission id.
+//      $trial_redirect_path = 'new_trial-trial/results';
+//
+//      // Machine name of field that should store the closest center found after form processing.
+//      $trial_location_field_name = 'field_new_trial_closest_center';
+//      break;
+  }
+
+  // Pass values to the form.
```

Comment:
The below code runs regardless of if this is a trial form. If we make the addition of the validation conditional on it actually being a trial form that won't be an issue. If we don't we could skip this code if we hit a the default case in the switch statement above.

Reply:
As I mentioned in my comment above, I like the default scenario here. To handle that I am introducing an `is_trial` flag and checking for its presence before executing this code.